### DISCUSSION TITLE
Identity and social key recovery (Reticulum Identities)

### DISCUSSION THREAD

**Discussion by samuk** [bluesky-social/atproto#363] [Protocol (atproto)]
I'm interested in identity: https://atproto.com/guides/identity

Background: Sideband is an existing Android app that creates Reticulum Identities. These are an X25519/Ed25519 dual keypair, with the X25519 keypair used for generating ephemeral session keys for links and such, and the Ed25519 pair used for signing and authentication purposes. https://github.com/markqvist/Reticulum/discussions/155#discussioncomment-4187299

https://darkcrystal.pw/threat-model/#introduction is an existing implementation of Shamir secret sharing.

Question: Could atproto use my Reticulum ID?

**Comment by dholms** [comment]
Our identity system is based on DIDs. Right now, we support did:plc & did:web. PLC cryptography is extensible, right now it supports secp256k1 & NIST P-256, but will likely support ed25519 in the future. Right now your PDS controls the signing key, but we may update this in the future to allow for user-held keys. There are a lot of UX & DX affordances to take into account before doing so. Reticulum may fit in for managing the recovery key for your DID. User-held recovery keys are your recourse against an adversarial PDS that acts in bad faith.

**Reply by dholms** [reply]
I guess to be more direct: our identity system is not fully swappable. We expect identities of a certain format. However, management of a recovery key is more flexible. The correct tooling would need to be built on Reticulum to allow for signing PLC txs, but it's possible

**Reply by samuk** [reply]
Thanks. I'll make the lead dev aware of that and see if he has any interest in becoming DID compatible.
